src/EPR.RegulatorService.Frontend.Web/Controllers/RegistrationSubmissions/RegistrationSubmissionsController.cs

```diff
@@ -165,23 +166,62 @@ public async Task<IActionResult> GrantRegistrationSubmission(Guid? organisationI
     {
         _currentSession = await _sessionManager.GetSessionAsync(HttpContext.Session);
 
-        if (!GetOrRejectProvidedOrganisationId(organisationId, out RegistrationSubmissionDetailsViewModel existingModel))
+        if (!GetOrRejectProvidedOrganisationId(organisationId, out var existingModel))
         {
             return RedirectToAction(PagePath.PageNotFound, "RegistrationSubmissions");
         }
 
-        SetBackLink(Url.RouteUrl("SubmissionDetails", new { organisationId }), false);
+        SetBackLink($"{PagePath.RegistrationSubmissionDetails}/{existingModel.OrganisationId}");
 
-        var model = new GrantRegistrationSubmissionViewModel()
+        var model = new GrantRegistrationSubmissionViewModel
         {
-            OrganisationId = organisationId.Value
+            OrganisationId = existingModel.OrganisationId
         };
 
         ViewBag.BackToAllSubmissionsUrl = Url.Action("RegistrationSubmissions");
 
         return View(nameof(GrantRegistrationSubmission), model);
     }
 
+    [HttpPost]
+    [Route(PagePath.GrantRegistrationSubmission + "/{organisationId:guid}", Name = "GrantRegistrationSubmission")]
+    public async Task<IActionResult> GrantRegistrationSubmission(GrantRegistrationSubmissionViewModel model)
+    {
+        _currentSession = await _sessionManager.GetSessionAsync(HttpContext.Session);
+
+        if (!GetOrRejectProvidedOrganisationId(model.OrganisationId, out var existingModel))
+        {
+            return RedirectToAction(PagePath.PageNotFound, "RegistrationSubmissions");
+        }
+
+        if (!ModelState.IsValid && model.IsGrantRegistrationConfirmed == null)
```

Comment:
Is the null check redundant?

Reply:
@shehzad-ismail8 too discussed the same thing with me and I have an opinion on this. Happy to discuss with you as well.

Reply:
Ah you both are right. I was overthinking on this I suppose. 
Thanks @shehzad-ismail8/ @JulianYoungEviden, thanks for raising this and sorry @shehzad-ismail8 for not changing this when you pointed out as  I was worrying about not missing any business rule/any further addition of another required property(s) in the model etc...